arch/arm64/core/cache.c

```diff
@@ -57,3 +62,65 @@ size_t arch_dcache_line_size_get(void)
 
 	return dcache_line_size;
 }
+
+
+/*
+ * operation for all data cache
+ * ops:  K_CACHE_INVD: invalidate
+ *	 K_CACHE_WB: clean
+ *	 K_CACHE_WB_INVD: clean and invalidate
+ */
+int arch_dcache_all(int op)
+{
+	uint32_t clidr_el1, csselr_el1, ccsidr_el1;
+	uint8_t loc, ctype, cache_level, line_size, way_pos;
+	uint32_t max_ways, max_sets, dc_val, set, way;
+
+	if (op != K_CACHE_INVD && op != K_CACHE_WB && op != K_CACHE_WB_INVD)
+		return -ENOTSUP;
+
+	clidr_el1 = read_sysreg(CLIDR_EL1);
```

Comment:
Instead of using the generic `read_sysreg` add a new register helper with `MAKE_REG_HELPER(clidr)` and use `{read,write}_clidr_el1()`. The same for all the other registers in this function.

Reply:
Done, thanks.